### PR TITLE
fix(scheduler-targets-alpha): add dlq policy to execution role instead of queue policy

### PR DESCRIPTION
**Pull request by gracelu0**
### Issue # (if applicable)

Tracking #31785.

### Reason for this change

Currently if a dead letter queue (DLQ) is specified then a queue policy is created for the DLQ which allows the schedule to send messages. This is incorrect and the permissions should be added to the schedule's execution role instead.  

### Description of changes

Add `sqs:SendMessage` permission to execution role's policy statement if dead letter queue is specified. This follows the [service docs](https://docs.aws.amazon.com/scheduler/latest/UserGuide/configuring-schedule-dlq.html#configuring-schedule-dlq-permissions) for configuring a schedule DLQ.

Also removed cross-region validation as the deployment will fail fast for this case so the validation is unnecessary.

### Description of how you validated changes

Updated unit tests and added a new integration test with dead letter queue setup on the schedule

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-scheduler-targets-alpha/lib/target.ts [hunk 1]
```diff
@@ -148,25 +148,14 @@ export abstract class ScheduleTargetBase {
...
-  private addToDeadLetterQueueResourcePolicy(schedule: ISchedule, queue: sqs.IQueue) {
-    if (!sameEnvDimension(schedule.env.region, queue.env.region)) {
-      throw new Error(`Cannot assign Dead Letter Queue in region ${queue.env.region} to the schedule ${Names.nodeUniqueId(schedule.node)} in region ${schedule.env.region}. Both the queue and the schedule must be in the same region.`);
-    }
-
+  private addDeadLetterQueueActionToRole(schedule: ISchedule, role: iam.IRole, queue: sqs.IQueue) {
     // Skip Resource Policy creation if the Queue is not in the same account.
     // There is no way to add a target onto an imported schedule, so we can assume we will run the following code only
     // in the account where the schedule is created.
     if (sameEnvDimension(schedule.env.account, queue.env.account)) {
```
Comment (samson-keung) [inline packages/@aws-cdk/aws-scheduler-targets-alpha/lib/target.ts:157]: This check is not needed. The queue and the schedule can be in different account and in such case, we still want to add the policy to the role.

File: packages/@aws-cdk/aws-scheduler-targets-alpha/test/codebuild-start-build.test.ts [hunk 2]
```diff
@@ -485,10 +466,26 @@ describe('codebuild start build', () => {
...
+            Resource: importedQueue.queueArn,
+          },
+        ],
+      },
+      Roles: [{ Ref: roleId }],
+    });
   });
 
-  test('does not create a queue policy when DLQ is created in a different account', () => {
+  test('does not add IAM policy when DLQ is created in a different account', () => {
```
Comment (samson-keung) [inline packages/@aws-cdk/aws-scheduler-targets-alpha/test/codebuild-start-build.test.ts:488]: This unit test should be deleted as described in comment above.

**Comment by samson-keung** [comment]
For integ test, you can write one similar to [integ.schedule.ts](https://github.com/samson-keung/aws-cdk/blob/09f7ad1696db0b12444ad0941714fa8f84110d58/packages/%40aws-cdk/aws-scheduler-alpha/test/integ.schedule.ts) but intentionally have missing target invocation permission in the execution role. A rate-based schedule without time window is invoked right away when created. Then use assert to check the queue has the expected message.

**Review by samson-keung — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/aws-scheduler-targets-alpha/lib/target.ts (hunk 1 above)
Comment (gracelu0) [reply packages/@aws-cdk/aws-scheduler-targets-alpha/lib/target.ts:157]: Ah I see, will remove

**Review by xazhao — APPROVED** [review]
LGTM

**Pull request merged**